pxr/base/tf/hash.h

```diff
@@ -33,6 +33,9 @@
 
 #include <cstring>
 #include <string>
+#include <set>
+#include <map>
+#include <typeindex>
```

Comment:
WBN to maintain headers in alphabetical order here and in the other files in this change.